### PR TITLE
Completed work on bug for Tomorrow.

### PR DESCRIPTION
**Pull request by ocampd1**

### CODE REVIEW

File: chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts [hunk 1]
```diff
@@ -18,14 +18,22 @@ import {LanguagePopoverComponent} from '../language-popover/language-popover.com
...
 export class GenerateDrawTimeCardComponent implements OnInit, OnDestroy {
+
+    constructor(private pick3WebScrappingService: Pick3WebScrapingProviderService,
+                public toastService: IonicToastNotificationService,
+                public drawStateService: DrawStateService) {
+        this.pick3StateLottery = pick3WebScrappingService.findRegisteredStateLottery('TX');
+    }
+
+    public currentDate = new Date().getDate();
+    defaultDrawingTimes = ['draw.time.enum.morning', 'draw.time.enum.day', 'draw.time.enum.evening', 'draw.time.enum.night'];
```
[the remark this answers is not in the document]
Comment (ocampd1) [reply chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts:29]: Replaced value of enums.

File: chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts [hunk 2]
```diff
@@ -18,14 +18,22 @@ import {LanguagePopoverComponent} from '../language-popover/language-popover.com
...
     @Input() defaultDrawDateTime: Pick3DrawTimeEnum.Pick3DrawTimeEnum;
 
     public showCountDownToDrawing = false;
 
-    timesNotAvailable = [];
-    timesToGenerate = [];
+    newDrawingTimes = [];
     drawTimes: Array<Pick3DrawTimeCard> = [
         new Pick3DrawTimeCardDomain({
             title: 'draw.time.enum.morning',
```
[the remark this answers is not in the document]
Comment (ocampd1) [reply chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts:39]: Created and used constants for values.

File: chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts [hunk 3]
```diff
@@ -83,6 +85,22 @@ export class GenerateDrawTimeCardComponent implements OnInit, OnDestroy {
         this.pick3StateLottery = null;
     }
 
+    public setDrawingTimeMenuItems(selectedDrawTime, timeOfDay) {
+        const drawingTimeMenuItem = [];
+        const date = new Date().getDate();
+
+        if (timeOfDay === date) {
+            this.resetDrawingTimes();
+            for (const i of this.drawTimes) {
```
[the remark this answers is not in the document]
Comment (ocampd1) [reply chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts:94]: renamed to drawTime.

File: chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts [hunk 4]
```diff
@@ -83,6 +85,22 @@ export class GenerateDrawTimeCardComponent implements OnInit, OnDestroy {
         this.pick3StateLottery = null;
     }
 
+    public setDrawingTimeMenuItems(selectedDrawTime, timeOfDay) {
```
[the remark this answers is not in the document]
Comment (ocampd1) [reply chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts:88]: Added proper method declarations.

**Review by ocampd1 — COMMENTED** [review]
Completed changes as requested!

File: chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts [hunk 5]
```diff
@@ -85,14 +91,14 @@ export class GenerateDrawTimeCardComponent implements OnInit, OnDestroy {
         this.pick3StateLottery = null;
     }
 
-    public setDrawingTimeMenuItems(selectedDrawTime, timeOfDay) {
+    public setDrawingTimeMenuItems(selectedDrawTime, timeOfDay): void {
```
Comment (pouncilt) [inline chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts:94]: User proper method declaration.  I.E  declare parameter types.

File: chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts [hunk 6]
```diff
@@ -207,7 +213,7 @@ export class GenerateDrawTimeCardComponent implements OnInit, OnDestroy {
         });
```
Comment (pouncilt) [inline chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts:213]: Looks like we don't need else block any more.

File: chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts [hunk 7]
```diff
@@ -275,25 +279,25 @@ export class GenerateDrawTimeCardComponent implements OnInit, OnDestroy {
     }
 
 
-    public selectYesterdayPrevDrawingDate(yesterday, today) {
+    public selectYesterdayPrevDrawingDate(yesterday, today): void {
```
Comment (pouncilt) [inline chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts:282]: User proper method declaration. I.E declare parameter types.

File: chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts [hunk 8]
```diff
@@ -275,25 +279,25 @@ export class GenerateDrawTimeCardComponent implements OnInit, OnDestroy {
...
         yesterday.style.backgroundColor = '#2fdf75';
         today.style.backgroundColor = '#e5e5e5';
     }
 
-    public selectTodayPrevDrawingDate(today, yesterday) {
+    public selectTodayPrevDrawingDate(today, yesterday): void {
```
Comment (pouncilt) [inline chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts:287]: User proper method declaration. I.E declare parameter types.

File: chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts [hunk 9]
```diff
@@ -275,25 +279,25 @@ export class GenerateDrawTimeCardComponent implements OnInit, OnDestroy {
...
         today.style.backgroundColor = '#2fdf75';
         yesterday.style.backgroundColor = '#e5e5e5';
     }
 
-    public selectTomorrowDrawingDate(tomorrow, today) {
+    public selectTomorrowDrawingDate(tomorrow, today): void {
```
Comment (pouncilt) [inline chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts:292]: User proper method declaration. I.E declare parameter types.

File: chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts [hunk 10]
```diff
@@ -275,25 +279,25 @@ export class GenerateDrawTimeCardComponent implements OnInit, OnDestroy {
...
         const tomorrowsDate = new Date();
         tomorrow.style.backgroundColor = '#2fdf75';
         today.style.backgroundColor = '#e5e5e5';
 
         this.setDrawingTimeMenuItems('today', tomorrowsDate.setDate(tomorrowsDate.getDate() + 1));
     }
 
-    public selectTodayDrawingDate(today, tomorrow) {
+    public selectTodayDrawingDate(today, tomorrow): void {
```
Comment (pouncilt) [inline chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts:300]: User proper method declaration. I.E declare parameter types.

File: chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts [hunk 11]
```diff
@@ -302,28 +306,28 @@ export class GenerateDrawTimeCardComponent implements OnInit, OnDestroy {
 
     }
 
-    public submitGenerate(generateDisplay, continueDisplay) {
+    public submitGenerate(generateDisplay, continueDisplay): void {
```
Comment (pouncilt) [inline chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts:309]: User proper method declaration. I.E declare parameter types.

File: chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts [hunk 12]
```diff
@@ -302,28 +306,28 @@ export class GenerateDrawTimeCardComponent implements OnInit, OnDestroy {
...
         continueDisplay.style.display = 'block';
         generateDisplay.style.display = 'none';
     }
 
-    public continueGenerate(continueDisplay, generateDisplay) {
+    public continueGenerate(continueDisplay, generateDisplay): void {
```
Comment (pouncilt) [inline chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts:314]: User proper method declaration. I.E declare parameter types.

File: chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts [hunk 13]
```diff
@@ -302,28 +306,28 @@ export class GenerateDrawTimeCardComponent implements OnInit, OnDestroy {
...
         continueDisplay.style.display = 'none';
         generateDisplay.style.display = 'block';
     }
 
-    public showSecondBackButton(secondBtn, firstBtn) {
+    public showSecondBackButton(secondBtn, firstBtn): void {
```
Comment (pouncilt) [inline chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts:319]: User proper method declaration. I.E declare parameter types.

File: chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts [hunk 14]
```diff
@@ -302,28 +306,28 @@ export class GenerateDrawTimeCardComponent implements OnInit, OnDestroy {
...
         secondBtn.style.display = 'block';
         firstBtn.style.display = 'none';
     }
 
-    public showFirstBackButton(firstBtn, secondBtn) {
+    public showFirstBackButton(firstBtn, secondBtn): void {
```
Comment (pouncilt) [inline chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts:324]: User proper method declaration. I.E declare parameter types.

File: chucks-pick3-mobile/src/app/models/pick3-draw-time.enum.ts [hunk 15]
```diff
@@ -1,45 +1,53 @@
...
+
                 break;
         }
 
-        return propertyKey
+        return propertyKey;
     }
 
-    export function toString(e: Pick3DrawTimeEnum|string): string {
+    export function toString(e: Pick3DrawTimeEnum | string): string {
```
Comment (pouncilt) [inline chucks-pick3-mobile/src/app/models/pick3-draw-time.enum.ts:39]: Nice!

**Review by pouncilt — APPROVED** [review]

File: chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts (hunk 9 above)
Comment (ocampd1) [reply chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts:292]: Resolved

File: chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts (hunk 10 above)
Comment (ocampd1) [reply chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts:300]: Resolved

File: chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts (hunk 11 above)
Comment (ocampd1) [reply chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts:309]: Resolved

File: chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts (hunk 12 above)
Comment (ocampd1) [reply chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts:314]: Resolved

File: chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts (hunk 13 above)
Comment (ocampd1) [reply chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts:319]: Resolved

File: chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts (hunk 14 above)
Comment (ocampd1) [reply chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts:324]: Resolved

File: chucks-pick3-mobile/src/app/models/pick3-draw-time.enum.ts (hunk 15 above)
Comment (ocampd1) [reply chucks-pick3-mobile/src/app/models/pick3-draw-time.enum.ts:39]: Thanks!

File: chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts (hunk 8 above)
Comment (ocampd1) [reply chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts:287]: Resolved

File: chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts (hunk 7 above)
Comment (ocampd1) [reply chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts:282]: Resolved

File: chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts (hunk 6 above)
Comment (ocampd1) [reply chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts:213]: Else block was not found in code.

File: chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts (hunk 5 above)
Comment (ocampd1) [reply chucks-pick3-mobile/src/app/components/generate-draw-time-card/generate-draw-time-card.component.ts:94]: Resolved and refactored code.

**Pull request merged**